cardano-cli/src/Cardano/CLI/Shelley/Run/Pretty.hs

```diff
@@ -1,107 +1,175 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 
-module Cardano.CLI.Shelley.Run.Pretty (prettyTx) where
+-- | User-friendly pretty-printing for textual user interfaces (TUI)
+module Cardano.CLI.Shelley.Run.Pretty (friendlyTxBodyBS) where
 
-import           Cardano.Api
-                   (ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley))
+import           Cardano.Prelude
+
+import           Cardano.Api as Api (AddressInEra (..),
+                   AddressTypeInEra (ByronAddressInAnyEra, ShelleyAddressInEra), CardanoEra,
+                   ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley),
+                   ShelleyEra, TxBody, serialiseAddress)
 import           Cardano.Api.Byron (TxBody (ByronTxBody))
-import           Cardano.Api.Shelley (TxBody (ShelleyTxBody))
+import           Cardano.Api.Shelley (TxBody (ShelleyTxBody), fromShelleyAddr)
+import           Cardano.Binary (Annotated)
 import           Cardano.CLI.Helpers (textShow)
+import qualified Cardano.Chain.UTxO as Byron
+import           Cardano.Ledger.Shelley as Ledger (ShelleyEra)
+import           Cardano.Ledger.ShelleyMA (MaryOrAllegra (Allegra, Mary), ShelleyMAEra)
 import qualified Cardano.Ledger.ShelleyMA.TxBody as ShelleyMA
-import           Cardano.Prelude
-import           Data.Aeson as JSON (Value, object, (.=))
-import           Data.Aeson.Encode.Pretty (Config (confCompare), defConfig, encodePretty')
+import           Data.Aeson (Object, Value (..), object, toJSON, (.=))
```

Comment:
Have a look at: https://github.com/input-output-hk/cardano-node/blob/master/cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs#L20

We tend to separate hackage packages from local packages.

Reply:
Oops, my bad. Sure.